src/imagings/requests/NewImagingRequest.tsx

```diff
@@ -46,10 +48,21 @@ const NewImagingRequest = () => {
   useAddBreadcrumbs(breadcrumbs)
 
   const onPatientChange = (patient: Patient) => {
-    setNewImagingRequest((previousNewImagingRequest) => ({
-      ...previousNewImagingRequest,
-      patient: patient.fullName as string,
-    }))
+    if (patient) {
+      setNewImagingRequest((previousNewImagingRequest) => ({
+        ...previousNewImagingRequest,
+        patient: patient.fullName as string,
+      }))
+
+      const visits = patient.visits?.map((v) => ({ label: v.type, value: v.id }))
+      setVisitOption(visits)
```

Comment:
I believe we need the visits to be a state variable. Because we can only fetch the visits data inside the `onPatientChange` method using the patient parameter (as we fetch the patient using this line `onSearch={async (query: string) => PatientRepository.search(query)}`). Using state variable ensures that the visits data will be re-rendered whenever the patient field value is changed.